modules/apps/object/object-rest-test/src/testFunctional/tests/RetrieveTheRelatedElementsOfSystemObject.testcase

```diff
@@ -0,0 +1,244 @@
+				objectDefinitionId1 = "${objectDefinitionId1}",
+				objectDefinitionId2 = "${objectDefinitionId2}",
+				type = "manyToMany");
+		}
+
+		task ("Given oneToMany relationship userSecondFoundations created") {
+			var relationshipName = ObjectDefinitionAPI.createRelationship(
+				deletionType = "cascade",
+				en_US_label = "UserSecondFoundations",
+				name = "userSecondFoundations",
+				objectDefinitionId1 = "${objectDefinitionId1}",
+				objectDefinitionId2 = "${objectDefinitionId3}",
+				type = "oneToMany");
+		}
+
+		task ("Given object entries created") {
+			ObjectDefinitionAPI.setUpGlobalObjectEntryId();
+		}
+
+		task ("Given two userAccount entries created") {
+			UserAccountAPI.setUpGlobalUserAccountId(
+				alternateName1 = "user1",
+				alternateName2 = "user2",
+				emailAddress1 = "user1@liferay.com",
+				emailAddress2 = "user2@liferay.com",
+				familyName1 = "userfn1",
+				familyName2 = "userfn2",
+				givenName1 = "usergn1",
+				givenName2 = "usergn2");
+		}
+	}
+
+	tearDown {
+		for (var relationshipName : list "usersFoundations,userSecondFoundations") {
+			ObjectDefinitionAPI.deleteRelationship(
+				name = "User",
+				relationshipName = "${relationshipName}");
+		}
+
+		for (var objectName : list "Foundation,SecondFoundation") {
+			ObjectAdmin.deleteObjectViaAPI(objectName = "${objectName}");
+		}
+
+		for (var userId : list "${staticUserAccountId1},${staticUserAccountId2}") {
+			JSONUser.deleteUserByUserId(userId = "${userId}");
+		}
+	}
+
+	@disable-webdriver = "true"
+	@priority = "3"
+	test CanGetRelationshipDetailsThroughRelationshipEndpointWithoutRelatedEntries {
+		property portal.acceptance = "true";
+
+		task ("When I call userAccount object GET endpoint with {userAccountId}/usersFoundations") {
+			UserAccountAPI.getRelationshipByUserAccountId(
+				relationshipName = "usersFoundations",
+				userAccountId = "${staticUserAccountId1}");
+		}
+
+		task ("Then I receive a correct response with GET endpoint {userAccount}/usersFoundations") {
+			UserAccountAPI.assertResponseNotIncludeDetailsOfDeletedObjectEntry(
+				customObjectEntryId = "${staticObjectEntryId1}",
+				expectedValue = "[]",
+				relationshipName = "usersFoundations",
+				userAccountId = "${staticUserAccountId1}");
+		}
+	}
+
+	@disable-webdriver = "true"
+	@priority = "3"
+	test CanNotGetDetailsAfterRelationshipIsDeleted {
+		property portal.acceptance = "true";
+
+		task ("And Given I relate the foundation entry with the userAccount entry through the userAccount PUT endpoint") {
+			UserAccountAPI.relateObjectEntries(
+				customObjectId = "${staticObjectEntryId3}",
+				relationshipName = "userSecondFoundations",
+				userAccountId = "${staticUserAccountId1}");
+		}
+
+		task ("When deleting the relationship") {
+			ObjectDefinitionAPI.deleteRelationship(
+				name = "User",
+				relationshipName = "userSecondFoundations");
+		}
+
+		task ("Then the GET endpoint {userAccountId}/usersFoundations is no longer available") {
```

Comment:
I think we can make the description in `Then` step and the codes to be consistent. I mean either use usersFoundations or use userSecondFoundations :smile:

Reply:
aren't these two different endpoints though? As I understand two relationships are being created here: usersFoundations and userSecondFoundations? 🤔

Reply:
Yes, they are two different endpoints. I mean in the task ("Then the GET endpoint {userAccountId}/**usersFoundations** is no longer available`), and in the code we testing the userSecondFoundations is deleted and no longer available, so maybe we can update to userSecondFoundations in the Then description?

Reply:
oh, now I understand :D Yes, in this case, it needs the THEN step needs to be aligned with what is exactly happening in the code. @Hijky please modify 🙏🏻